### ISSUE TITLE
NameError when using RubyGems created command

### ISSUE THREAD

**Issue by trnubo** [sensu-plugins/sensu-plugins-ponymailer#3]
I'm getting the following error when running `/opt/sensu/embedded/bin/handler-ponymailer.rb`

```
/opt/sensu/embedded/lib/ruby/2.0.0/net/protocol.rb:52:in `<module:Net>': uninitialized constant Timeout::Error (NameError)
        from /opt/sensu/embedded/lib/ruby/2.0.0/net/protocol.rb:24:in `<top (required)>'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/2.0.0/net/http.rb:22:in `<top (required)>'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/gems/2.0.0/gems/sensu-plugin-1.2.0/lib/sensu-handler.rb:1:in `<top (required)>'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/gems/2.0.0/gems/sensu-plugins-ponymailer-0.0.3/bin/handler-ponymailer.rb:12:in `<top (required)>'
        from /opt/sensu/embedded/bin/handler-ponymailer.rb:23:in `load'
        from /opt/sensu/embedded/bin/handler-ponymailer.rb:23:in `<main>'
```

However when I run `/opt/sensu/embedded/lib/ruby/gems/2.0.0/gems/sensu-plugins-ponymailer-0.0.3/bin/handler-ponymailer.rb` the command is successful. 

```
mail -- sent alert for localhost.localdomain/keepalive to ...
```

I'm testing by piping the following to the commands

```
{"id":"7f21d4dc-02aa-4c57-8282-a2e8d0b20565","client":{"name":"localhost.localdomain","address":"localhost.localdomain","subscriptions":["test"],"version":"0.20.1","timestamp":1438148285},"check":{"thresholds":{"warning":120,"critical":180},"handler":"keepalive","name":"keepalive","issued":1438148438,"executed":1438148438,"output":"No keepalive sent from client for 153 seconds (>=120)","status":1,"history":["0","0","0","0","0","1","1"],"total_state_change":0},"occurrences":1,"action":"create","timestamp":1438148438}
```

The contents of `/opt/sensu/embedded/bin/handler-ponymailer.rb` are

```
#!/opt/sensu/embedded/bin/ruby
#
# This file was generated by RubyGems.
#
# The application 'sensu-plugins-ponymailer' is installed as part of a gem, and
# this file is here to facilitate running it.
#

require 'rubygems'

version = ">= 0"

if ARGV.first
  str = ARGV.first
  str = str.dup.force_encoding("BINARY") if str.respond_to? :force_encoding
  if str =~ /\A_(.*)_\z/
    version = $1
    ARGV.shift
  end
end

gem 'sensu-plugins-ponymailer', version
load Gem.bin_path('sensu-plugins-ponymailer', 'handler-ponymailer.rb', version)
```

This is all in a debian:jessie docker image with the sensu 0.20.1-1 package installed. I've been working around the issue by using the long path to the script.

I'm happy to try debug this but have no idea how to go about this one. Its beyond my basic understanding of ruby.

**Comment by mattyjones** [comment]
@trnubo  I have no idea upon a first pass but I will try and take a look when I get some time.

**Comment by eheydrick** [comment]
This looks like the timeout gem vs stdlib timeout issue. The timeout gem is just a stub to install timeout-extensions but it conflicts with the built-in timeout. We should either switch to timeout-extensions or use the built-in timeout. Same issue in https://github.com/sensu-plugins/sensu-plugins-fluentd/issues/1

**Comment by eheydrick** [comment]
Fixed in version [0.0.4](https://rubygems.org/gems/sensu-plugins-ponymailer/versions/0.0.4).

**Comment by trnubo** [comment]
Thanks @eheydrick. Can confirm version 0.0.4 is working.
